lab_3_keywords_textrank/main.py

```diff
@@ -302,14 +353,26 @@ def fill_positions(self, tokens: tuple[int, ...]) -> None:
             tokens : tuple[int, ...]
                 sequence of tokens
         """
-        pass
+        for idx, token in enumerate(tokens):
+            if tokens[idx] not in self._positions:
+                self._positions[tokens[idx]] = []
+            self._positions[tokens[idx]] += [idx + 1]
 
     # Step 8.3
     def calculate_position_weights(self) -> None:
         """
         Computes position weights for all tokens in text
         """
-        pass
+        non_norm_weight = 0
+        non_norm_total_weight = 0
+        for token, positions in self._positions.items():
+            for position in positions:
+                non_norm_weight += 1 / position
+            self._position_weights[token] = non_norm_weight
+            non_norm_total_weight += non_norm_weight
+
+        for token in self._position_weights:
+            self._position_weights[token] = self._position_weights.get(token) / non_norm_total_weight
```

Comment:
is there a chance that `self._position_weights.get(token)` will return `None`?

Reply:
no, cause we added every token from `self._positions` to `self._position_weights` previously:
`for token, positions in self._positions.items():
      ...
      self._position_weights[token] = non_norm_weight`
is that right?

Reply:
True!